docs/adr/0002-matching-adyen-notification.md

```diff
@@ -25,3 +25,32 @@ by `interactionId` where `interactionId=${pspReference}`.
 - It is easier to fetch a key rather than using a custom field, also a key is an indexed field, so with a key, it's more performant.
 - The payment key is unique for all payments.
 - It's not possible to set key with my-payments endpoint. This prevents by default changing/removing the key accidentally. It is more secure than custom fields as the custom field might be changed with my-payment endpoint. Check for more details: https://docs.commercetools.com/api/projects/me-payments
+
+
+Date: 2023-01-03
+
+## Status
+
+[Accepted](https://github.com/commercetools/commercetools-adyen-integration/pull/1049)
+
+## Context
+
+The Adyen notification needs to be matched by its commercetools payment equivalent.
+We are using the payment key for the merchantReference and fetching the commercetools payment object with query `key="${merchantReference}"`.
+Since we have introduced custom reference for refund 
+in [v9.10.0](https://github.com/commercetools/commercetools-adyen-integration/releases/tag/v9.10.0), payment is no long able to be obtained by merchant reference as key. 
+
+The alternative for that is the native payment `key` field.
```

Comment:
What does it mean `native`? Is there any `non-native` payment? I think it's better to use `commercetools` instead of `native`.

Reply:
We were using payment key before as well. See the previous ADR. So this is not the main point here.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/1049/commits/8a7c1a769eea732f9b4cce72d1f8cd6ad4bd7b02